src/test/userDb.test.js

```diff
@@ -0,0 +1,195 @@
+import {
+  collection,
+  addDoc,
+  getDocs,
+  setDoc,
+  where,
+  query,
+  doc,
+} from "firebase/firestore";
+
+import {
+  getAllUsers,
+  getUser,
+  postUser,
+  updateUser,
+  postMockProperty,
+  postMockUser,
+} from "@/firebase/userDb";
+
+import { db } from "@/firebase/firebaseConfig";
+
+const tableName = "users";
+jest.mock("firebase/firestore", () => ({
+  setDoc: jest.fn(),
+  getFirestore: jest.fn(),
+  collection: jest.fn(),
+  addDoc: jest.fn(),
+  query: jest.fn(),
+  getDocs: jest.fn(),
+  where: jest.fn(),
+  doc: jest.fn(),
+}));
+
+describe("User Database", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("Get All the user.", async () => {
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              uid: "1231",
+              firstName: "Walter",
+              lastName: "White",
+              gender: "Female",
+            },
+            {
+              uid: "1234",
+              firstName: "Joe",
+              lastName: "White",
+              gender: "Male",
+            },
+          ],
+        });
+      },
+    };
+    getDocs.mockImplementation(() => querySnapshot);
+
+    // Call the tested function.
+    let testReturn = await getAllUsers();
+    console.log(testReturn);
+    expect(getDocs).toHaveBeenCalledWith(collection(tableName));
+    expect(collection).toHaveBeenCalledWith(tableName);
+    expect(testReturn).toEqual([
+      [
+        {
+          uid: "1231",
+          firstName: "Walter",
+          lastName: "White",
+          gender: "Female",
+        },
+        {
+          uid: "1234",
+          firstName: "Joe",
+          lastName: "White",
+          gender: "Male",
+        },
+      ],
+    ]);
+  });
+
+  it("Get user Information from userID.", async () => {
+    const userID = "1234";
+    query.mockImplementationOnce(() => ({
+      withConverter: jest.fn(),
+    }));
+    const querySnapshot = {
+      forEach: (fn) => {
+        fn({
+          data: () => [
+            {
+              uid: "1231",
+              firstName: "Walter",
+              lastName: "White",
+              gender: "Female",
+            },
+          ],
+        });
+      },
+    };
```

Comment:
same comment as in propertyDb, can you use the database from the earlier test case so we have a user *without* the user id in the database?